tests/integration/functional/pipelines/test_pipeline_context.py

```diff
@@ -97,3 +100,46 @@ def test_that_argument_as_get_artifact_of_model_version_in_pipeline_context_fail
     producer_pipe(False)
     with pytest.raises(RuntimeError):
         consumer_pipe()
+
+
+@step
+def producer() -> Annotated[str, "bar"]:
+    """Produce artifact with metadata and attach metadata to model version."""
+    ver = get_step_context().model_version.version
+    log_model_version_metadata(metadata={"foobar": "model_meta_" + ver})
+    log_artifact_metadata(metadata={"foobar": "artifact_meta_" + ver})
+    return "artifact_data_" + ver
+
+
+@step
+def asserter(artifact: str, artifact_metadata: str, model_metadata: str):
+    """Assert that passed in values are loaded in lazy mode, since they should not exists on compose run."""
+    ver = get_step_context().model_version.version
+    assert artifact == "artifact_data_" + ver
+    assert artifact_metadata == "artifact_meta_" + ver
+    assert model_metadata == "model_meta_" + ver
```

Comment:
is the idea that we're going to use these more in the future / going forward elsewhere?

Reply:
Not necessarily, but steps cannot be defined as inner functions, cause it messes up with the way we refer to the sources on step execution later on.

Reply:
Ah right yes. got it.